diff -rupN ori/eglibc-linaro-2.18-2013.10/sunrpc/Makefile eglibc-linaro-2.18-2013.10/sunrpc/Makefile
--- ori/eglibc-linaro-2.18-2013.10/sunrpc/Makefile	2013-10-08 12:22:54.000000000 +0200
+++ eglibc-linaro-2.18-2013.10/sunrpc/Makefile	2014-08-24 11:05:11.000000000 +0200
@@ -180,7 +180,7 @@ $(cross-rpcgen-objs): $(objpfx)cross-%.o
 		$(OUTPUT_OPTION) $(native-compile-mkdep-flags) -c
 
 $(objpfx)cross-rpcgen: $(cross-rpcgen-objs)
-	$(BUILD_CC) $^ $(BUILD_LDFLAGS) -o $@
+	$(BUILD_CC) $^ $(BUILD_LDFLAGS) -o $@ -L/usr/local/lib -lintl
 
 # This makes sure -DNOT_IN_libc is passed for all these modules.
 cpp-srcs-left := $(rpcgen-objs:.o=.c)
diff -rupN ori/eglibc-linaro-2.18-2013.10/sunrpc/rpc/types.h eglibc-linaro-2.18-2013.10/sunrpc/rpc/types.h
--- ori/eglibc-linaro-2.18-2013.10/sunrpc/rpc/types.h	2013-10-08 12:22:54.000000000 +0200
+++ eglibc-linaro-2.18-2013.10/sunrpc/rpc/types.h	2014-08-24 11:04:31.000000000 +0200
@@ -69,6 +69,9 @@ typedef unsigned long rpcport_t;
 #include <sys/types.h>
 #endif
 
+#define __u_char_defined
+#define __daddr_t_defined
+
 #ifndef __u_char_defined
 typedef __u_char u_char;
 typedef __u_short u_short;
