diff -ur orig/binutils/addr2line.c new/binutils/addr2line.c
--- orig/binutils/addr2line.c	2012-07-25 09:06:56.000000000 +1200
+++ new/binutils/addr2line.c	2012-12-17 15:56:44.827596232 +1300
@@ -389,6 +389,7 @@
   return 0;
 }
 
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -403,7 +404,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = *argv;
diff -ur orig/binutils/ar.c new/binutils/ar.c
--- orig/binutils/ar.c	2012-06-30 00:59:49.000000000 +1200
+++ new/binutils/ar.c	2012-12-17 15:56:44.827596232 +1300
@@ -650,7 +650,7 @@
 
   xexit (status);
 }
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int main (int, char **);
 
 int
@@ -668,7 +668,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = argv[0];
diff -ur orig/binutils/coffdump.c new/binutils/coffdump.c
--- orig/binutils/coffdump.c	2011-10-14 04:33:32.000000000 +1300
+++ new/binutils/coffdump.c	2012-12-17 15:56:44.827596232 +1300
@@ -472,7 +472,7 @@
 
   exit (status);
 }
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int ac, char **av)
 {
@@ -494,7 +494,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = av[0];
diff -ur orig/binutils/cxxfilt.c new/binutils/cxxfilt.c
--- orig/binutils/cxxfilt.c	2007-11-26 14:53:59.000000000 +1300
+++ new/binutils/cxxfilt.c	2012-12-17 15:56:44.831596254 +1300
@@ -167,7 +167,7 @@
 }
 
 extern int main (int, char **);
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -178,6 +178,39 @@
   program_name = argv[0];
   xmalloc_set_program_name (program_name);
 
+#if defined (HAVE_SETLOCALE) && defined (HAVE_LC_MESSAGES)
+  setlocale (LC_MESSAGES, "");
+#endif
+#if defined (HAVE_SETLOCALE)
+  setlocale (LC_CTYPE, "");
+#endif
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+
+      free (prefix_dir);
+    }
+  textdomain (PACKAGE);
+
   expandargv (&argc, &argv);
 
   while ((c = getopt_long (argc, argv, "_hinps:tv", long_options, (int *) 0)) != EOF)
diff -ur orig/binutils/dlltool.c new/binutils/dlltool.c
--- orig/binutils/dlltool.c	2012-05-05 15:05:26.000000000 +1200
+++ new/binutils/dlltool.c	2012-12-17 15:56:44.831596254 +1300
@@ -4019,7 +4019,7 @@
 };
 
 int main (int, char **);
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int ac, char **av)
 {
@@ -4035,7 +4035,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   expandargv (&ac, &av);
diff -ur orig/binutils/dllwrap.c new/binutils/dllwrap.c
--- orig/binutils/dllwrap.c	2012-02-09 17:51:44.000000000 +1300
+++ new/binutils/dllwrap.c	2012-12-17 15:56:44.835596279 +1300
@@ -601,7 +601,7 @@
 };
 
 int main (int, char **);
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -634,7 +634,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   expandargv (&argc, &argv);
diff -ur orig/binutils/elfedit.c new/binutils/elfedit.c
--- orig/binutils/elfedit.c	2012-02-09 17:51:43.000000000 +1300
+++ new/binutils/elfedit.c	2012-12-17 15:56:44.835596279 +1300
@@ -651,7 +651,7 @@
     fprintf (stream, _("Report bugs to %s\n"), REPORT_BUGS_TO);
   exit (exit_status);
 }
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char ** argv)
 {
@@ -663,7 +663,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   expandargv (&argc, &argv);
diff -ur orig/binutils/nlmconv.c new/binutils/nlmconv.c
--- orig/binutils/nlmconv.c	2012-02-09 17:51:43.000000000 +1300
+++ new/binutils/nlmconv.c	2012-12-17 15:56:44.835596279 +1300
@@ -150,7 +150,7 @@
 #endif
 
 /* The main routine.  */
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -208,7 +208,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = argv[0];
diff -ur orig/binutils/nm.c new/binutils/nm.c
--- orig/binutils/nm.c	2012-06-30 04:55:15.000000000 +1200
+++ new/binutils/nm.c	2012-12-17 15:56:44.835596279 +1300
@@ -1516,6 +1516,7 @@
     }
 }
 
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -1529,7 +1530,31 @@
   setlocale (LC_CTYPE, "");
   setlocale (LC_COLLATE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = *argv;
diff -ur orig/binutils/objcopy.c new/binutils/objcopy.c
--- orig/binutils/objcopy.c	2012-05-09 05:49:36.000000000 +1200
+++ new/binutils/objcopy.c	2012-12-17 15:56:44.835596279 +1300
@@ -4093,7 +4093,7 @@
 
   return 0;
 }
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char *argv[])
 {
@@ -4103,7 +4103,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = argv[0];
diff -ur orig/binutils/objdump.c new/binutils/objdump.c
--- orig/binutils/objdump.c	2012-10-22 23:25:11.000000000 +1300
+++ new/binutils/objdump.c	2012-12-17 15:56:44.839596294 +1300
@@ -3409,6 +3409,7 @@
   bfd_close (file);
 }
 
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -3423,7 +3424,31 @@
   setlocale (LC_CTYPE, "");
 #endif
 
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = *argv;
diff -ur orig/binutils/readelf.c new/binutils/readelf.c
--- orig/binutils/readelf.c	2012-11-06 05:27:35.000000000 +1300
+++ new/binutils/readelf.c	2012-12-17 15:56:44.839596294 +1300
@@ -13766,7 +13766,7 @@
   print_address (addr, stderr);
 }
 #endif
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char ** argv)
 {
@@ -13778,7 +13778,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   expandargv (&argc, &argv);
diff -ur orig/binutils/size.c new/binutils/size.c
--- orig/binutils/size.c	2009-09-11 01:40:44.000000000 +1200
+++ new/binutils/size.c	2012-12-17 15:56:44.839596294 +1300
@@ -117,7 +117,7 @@
 };
 
 int main (int, char **);
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -130,7 +130,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = *argv;
diff -ur orig/binutils/srconv.c new/binutils/srconv.c
--- orig/binutils/srconv.c	2011-03-01 07:32:51.000000000 +1300
+++ new/binutils/srconv.c	2012-12-17 15:56:44.839596294 +1300
@@ -1743,7 +1743,7 @@
     fprintf (ffile, _("Report bugs to %s\n"), REPORT_BUGS_TO);
   exit (status);
 }
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int ac, char **av)
 {
@@ -1767,7 +1767,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = av[0];
diff -ur orig/binutils/strings.c new/binutils/strings.c
--- orig/binutils/strings.c	2012-02-09 17:51:44.000000000 +1300
+++ new/binutils/strings.c	2012-12-17 15:56:44.839596294 +1300
@@ -134,7 +134,7 @@
 static long get_char (FILE *, file_ptr *, int *, char **);
 
 int main (int, char **);
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -147,7 +147,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_ALL, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = argv[0];
diff -ur orig/binutils/sysdump.c new/binutils/sysdump.c
--- orig/binutils/sysdump.c	2011-10-14 04:33:33.000000000 +1300
+++ new/binutils/sysdump.c	2012-12-17 15:56:44.843596312 +1300
@@ -653,7 +653,7 @@
     fprintf (ffile, _("Report bugs to %s\n"), REPORT_BUGS_TO);
   exit (status);
 }
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int ac, char **av)
 {
@@ -672,7 +672,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = av[0];
diff -ur orig/binutils/windmc.c new/binutils/windmc.c
--- orig/binutils/windmc.c	2011-06-03 01:43:18.000000000 +1200
+++ new/binutils/windmc.c	2012-12-17 15:56:44.843596312 +1300
@@ -933,7 +933,7 @@
 }
 
 int main (int, char **);
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -948,7 +948,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = argv[0];
diff -ur orig/binutils/windres.c new/binutils/windres.c
--- orig/binutils/windres.c	2012-02-09 17:51:44.000000000 +1300
+++ new/binutils/windres.c	2012-12-17 15:56:44.843596312 +1300
@@ -785,7 +785,7 @@
 int main (int, char **);
 
 /* The main function.  */
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -809,7 +809,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+          
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = argv[0];
diff -ur orig/gas/as.c new/gas/as.c
--- orig/gas/as.c	2012-06-08 00:47:23.000000000 +1200
+++ new/gas/as.c	2012-12-17 15:56:44.843596312 +1300
@@ -1126,7 +1126,7 @@
 }
 #endif
 
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char ** argv)
 {
@@ -1142,7 +1142,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   if (debug_memory)
diff -ur orig/gold/main.cc new/gold/main.cc
--- orig/gold/main.cc	2012-03-22 08:02:21.000000000 +1300
+++ new/gold/main.cc	2012-12-17 15:56:44.843596312 +1300
@@ -50,6 +50,12 @@
 #include "gdb-index.h"
 #include "timer.h"
 
+#ifdef _WIN32
+#include <io.h>
+#else
+#include <unistd.h>
+#endif
+
 using namespace gold;
 
 // This function emits the commandline to a hard-coded file in temp.
@@ -133,7 +137,7 @@
 
 #endif // !defined(DEBUG)
 
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main(int argc, char** argv)
 {
@@ -143,7 +147,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale(LC_CTYPE, "");
 #endif
-  bindtextdomain(PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = (char*)xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+
+      free (prefix_dir);
+    }
   textdomain(PACKAGE);
 
   program_name = argv[0];
Only in orig/gold: options.cc.orig
diff -ur orig/gprof/gprof.c new/gprof/gprof.c
--- orig/gprof/gprof.c	2009-06-13 03:33:30.000000000 +1200
+++ new/gprof/gprof.c	2012-12-17 15:56:44.843596312 +1300
@@ -176,7 +176,7 @@
   done (status);
 }
 
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -191,7 +191,31 @@
   setlocale (LC_CTYPE, "");
 #endif
 #ifdef ENABLE_NLS
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 #endif
 
diff -ur orig/ld/ldmain.c new/ld/ldmain.c
--- orig/ld/ldmain.c	2012-07-14 01:20:26.000000000 +1200
+++ new/ld/ldmain.c	2012-12-17 15:56:44.843596312 +1300
@@ -185,7 +185,7 @@
   (*default_bfd_assert_handler) (fmt, bfdver, file, line);
   config.make_executable = FALSE;
 }
-
+#define LIBINTL_RELATIVE_DIR "../share/locale"
 int
 main (int argc, char **argv)
 {
@@ -198,7 +198,31 @@
 #if defined (HAVE_SETLOCALE)
   setlocale (LC_CTYPE, "");
 #endif
-  bindtextdomain (PACKAGE, LOCALEDIR);
+  if (!access (LOCALEDIR, X_OK))
+    {
+      /* If LOCALEDIR exists, use LOCALEDIR. */
+      bindtextdomain(PACKAGE, LOCALEDIR);
+    }
+  else
+    {
+      /* Try relative dir, i.e.  .../bin/../share/locale. */
+      int len1, len2;
+      char *prefix_dir, *locale_dir;
+      prefix_dir = make_relative_prefix(argv[0],".",".");
+      len1 = strlen (prefix_dir);
+      len2 = strlen (LIBINTL_RELATIVE_DIR);
+      locale_dir = xmalloc (len1 + len2 + 1);
+      if (locale_dir != NULL)
+        {
+          strcpy (locale_dir, prefix_dir);
+          strcpy (locale_dir + len1, LIBINTL_RELATIVE_DIR);
+          bindtextdomain (PACKAGE, locale_dir);
+        }
+      else
+        bindtextdomain (PACKAGE, LOCALEDIR);
+
+      free (prefix_dir);
+    }
   textdomain (PACKAGE);
 
   program_name = argv[0];
